JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -468,81 +400,92 @@ private fun InterestsTabRowContent(
+            } else {
+                outerConstraints.copy(maxWidth = itemWidth)
+            }
+            val placeable = measureable.measure(itemConstraints)
+            // Update the height for each row
+            val row = index.floorDiv(columns)
+            rowHeights[row] = max(rowHeights[row], placeable.height)
+            placeable
+        }
 
-    val peopleSection = TabContent(Sections.People) {
-        val selectedPeople by interestsViewModel.selectedPeople.collectAsState()
-        PeopleList(
-            people = uiState.people,
-            selectedPeople = selectedPeople,
-            onPersonSelect = { interestsViewModel.togglePersonSelected(it) }
-        )
-    }
+        // Calculate maxHeight of the Interests layout. Heights of the row + top padding
+        val layoutHeight = topPaddingPx + rowHeights.sumOf { it }
+        // Calculate maxWidth of the Interests layout
+        val layoutWidth = itemWidth * columns + (itemSpacingPx * (columns - 1))
+        // Lay out given the max width and height
+        layout(
+            outerConstraints.maxWidth.coerceIn(0, layoutWidth),
+            outerConstraints.maxHeight.coerceIn(0, layoutHeight)
+        ) {
+            // Track the y co-ord we have placed children up to
+            var yPosition = topPaddingPx
+            placeables.forEachIndexed { index, placeable ->
+                // If there's only one column, then place items one after the other
+                if (columns == 1) {
+                    // Position item on the screen
+                    placeable.placeRelative(x = 0, y = yPosition)
+                    // Increment the y co-ord with the item's height
+                    yPosition += placeable.height
+                } else {
+                    // If there are multiple columns, then place items relative to each other
+                    if (index % columns == 0) {
+                        // Position item on a new row
+                        placeable.placeRelative(x = 0, y = yPosition)
+                    } else {
+                        // Place element in the same row, the previous item exists at this point
+                        val previousPlaceable = placeables[index - 1]
+                        placeable.placeRelative(
+                            x = previousPlaceable.width + itemSpacingPx,
+                            y = yPosition
+                        )
 
-    val publicationSection = TabContent(Sections.Publications) {
-        val selectedPublications by interestsViewModel.selectedPublications.collectAsState()
-        PublicationList(
-            publications = uiState.publications,
-            selectedPublications = selectedPublications,
-            onPublicationSelect = { interestsViewModel.togglePublicationSelected(it) }
-        )
+                        // Increment the y co-ord with the item's height
+                        yPosition += rowHeights[index.floorDiv(columns)]
```

Comment:
For a general number of columns, we should only be incrementing the `yPosition` when `index + 1 % columns == 0` (in other words, this is the last item in the row).

This works for 2 columns, since we only visit the `else` once for each row, but it wouldn't for 3 or more columns.

Reply:
Good point, this can also simplify the logic, thanks